structuralcodes/codes/mc2010/_concrete_shear.py

```diff
@@ -0,0 +1,1598 @@
+"""A collection of shear formulas for concrete"""
+import warnings
+from math import pi, tan, sin, cos
+
+
+def epsilon_x(
+    E_s: float,
+    As: float,
+    Med: float,
+    Ved: float,
+    Ned: float,
+    z: float,
+    delta_e: float,
+) -> float:
+    """The maximum allowed shear resistance
+
+    fib Model Code 2010, eq. (7.3-16)
+
+    Args:
+        E (float): The E-modulus to the material in MPa
+        As (Float): The cross-section area of reinforcement in mm^2
+        Med (Float): The moment working on the material in Nmm
+        Ved (float): The shear force working on the material in N
+        Ned: (float): The normal force working on the material in N
+        z: (float): The effective shear depth in mm
+        delta_E (float): The exentricity of the load in mm
+    Returns:
+        float: The longitudinal strain"""
+    return max(
+        (
+            (1 / (2 * E_s * As))
+            * (
+                (abs(Med) / z)
+                + abs(Ved)
+                + abs(Ned) * ((1 / 2) + (delta_e / z))
+            )
+        ),
+        0,
+    )
+
+
+def v_rd(
+    approx_lvl_c: int,
+    approx_lvl_s: int,
+    reinforcment: bool,
+    fck: float,
+    z: float,
+    bw: float,
+    dg: float,
+    E_s: float,
+    As: float,
+    Med: float,
+    Ved: float,
+    Ned: float,
+    delta_e: float,
+    alfa: float,
+    gamma_c: float,
+    asw: float,
+    sw: float,
+    f_ywd: float,
+    theta: float,
+) -> float:
+    """Compute the shear resistance of a web or slab.
+
+    fib Model Code 2010, Eq. (7.3-11)
+
+    Args:
+        approx_lvl_c (int): Approximation level for concrete
+        approx_lvl_s (int): Approximation level for reinforcment
+        reinforcment (bool): Shear reinforced concrete or no shear
+        reinforcement
+        fck (float): Characteristic strength in MPa
+        z: (float): distances between the centerline of the
+        compressive chord and the reinforcement in mm
+        bw: (float): Thickness of web in cross section in mm
+        dg: (float): Maximum size of aggregate in mm
+        E_s: (float): The E_s-modulus to the materialb in MPa
+        As: (float): The cross-section area of reinforcement in mm^2
+        Med: (float): The moment working on the material in Nmm
+        Ved: (float): The shear working on the material in N
+        Ned: (float): The normal force working on the material in N
+        delta_e (float): The exentricity of the load in mm
+        alfa (float): Inclination of the stirrups in degrees
+        gamma_c (float): Concrete safety factor
```

Comment:
Since there are also other `alpha`s in the Model Code, e.g. alpha_cc for calculating the design compressive strength, having gamma_c following directly behind the alpha inclination of the stirrups might be misleading. Consider moving gamma_c up below fck.